packaging/hudi-spark-bundle/pom.xml

```diff
@@ -66,10 +66,9 @@
                   <include>org.apache.hudi:hudi-common</include>
                   <include>org.apache.hudi:hudi-client-common</include>
                   <include>org.apache.hudi:hudi-spark-client</include>
-                  <include>org.apache.hudi:hudi-spark-common</include>
+                  <include>org.apache.hudi:hudi-spark-common_${scala.binary.version}</include>
                   <include>org.apache.hudi:hudi-spark_${scala.binary.version}</include>
-                  <include>org.apache.hudi:hudi-spark2_${scala.binary.version}</include>
-                  <include>org.apache.hudi:hudi-spark3_2.12</include>
+                  <include>org.apache.hudi:${hudi.spark.module}_${scala.binary.version}</include>
```

Comment:
so we had designed the bundle such that the hudi-spark2 and hudi-spark3 just need to be compiled differently, but included into a single bundle, and it works because we load up one of those dynamically based on spark version. Is that not working somehow?

Reply:
It works well for spark2 & spark3 currently when we only compile the right spark module  to the bundle jar. But if we compile both the spark2 and spark3 to the bundle jar as we do before, it will lead to some class conflict.

Reply:
okay. lets land this and tackle this after